content_delegate.js

```diff
@@ -227,11 +227,7 @@ ContentDelegate.prototype.handleSingleDocumentPageCheck = function() {
 ContentDelegate.prototype.onDocumentViewSubmit = function (event) {
   // Save a copy of the page source, altered so that the "View Document"
   // button goes forward in the history instead of resubmitting the form.
-  let originalSubmit = document.forms[0].getAttribute('onsubmit');
-  document.forms[0].setAttribute('onsubmit',
-				 'history.forward(); return false;');
-  let previousPageHtml = document.documentElement.innerHTML;
-  document.forms[0].setAttribute('onsubmit', originalSubmit);
+  let form = document.getElementById(event.data.id);
```

Comment:
To feel confident that this is right, I'd want to know that `document.forms[0]` and `document.getElementById(event.data.id)` resolve to the same form. I don't have that confidence from inspecting the code. And when I look at PACER forms with the DOM inspector, etc., I don't see an `id` property. And when I set a breakpoint at the top of `onDocumentViewSubmit()`, I also don't get that.

For the Bankruptcy case, which I guess is the brokenness this is trying to solve, they are certainly not the same:

```
>> event
message { target: Window → 032026507697, isTrusted: false, data: Object, origin: "https://ecf.caeb.uscourts.gov", lastEventId: "", source: Window → 032026507697, ports: Object, currentTarget: Window → 032026507697, eventPhase: 2, bubbles: false, cancelable: false }
>> event.data
Object { id: "form1516491714124" }
>> event.data.id
"form1516491714124"
 form = document.getElementById(event.data.id);
<form action="/doc1/032126507697" method="post" id="form1516491714124">
>> form.outerHTML
"<form action="/doc1/032126507697" method="post" id="form1516491714124"><input name="caseid" value="605035" type="hidden"><input name="got_receipt" value="1" type="hidden"></form>"
>> document.forms
HTMLCollection [ <form>, <form#form1516491714124> ]
>> document.forms.length
2
>> document.forms[0]
<form method="POST" action="https://ecf.caeb.uscourts.gov/doc1/032126507697" onsubmit="goDLS('/doc1/032126507697','605035','','1','','','','','','');return(false)">
>> document.forms[0].innerHTML
"<hr><center><table width="400" border="1" bgcolor="white"><tbody><tr><th colspan="4"><font size="+1" color="DARKRED">PACER Service Center </font></th></tr><tr><th colspan="4"><font color="DARKBLUE">Transaction Receipt </font></th></tr><tr></tr><tr></tr><tr><td colspan="4" align="CENTER"><font size="-1" color="DARKBLUE">Sat Jan 20 15:41:41 2018</font></td></tr><tr><th align="LEFT"><font size="-1" color="DARKBLUE"> Pacer Login: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> jhawkinson0:3355704:0 </font></td><th align="LEFT"><font size="-1" color="DARKBLUE"> Client Code: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE">  </font></td></tr><tr><th align="LEFT"><font size="-1" color="DARKBLUE"> Description: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> Image:72-0 </font></td><th align="LEFT"><font size="-1" color="DARKBLUE"> Case Number: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> 17-13797                       </font></td></tr><tr><th align="LEFT"><font size="-1" color="DARKBLUE"> Billable Pages: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> 2 </font></td><th align="LEFT"><font size="-1" color="DARKBLUE"> Cost: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> 0.20 </font></td></tr><tr></tr><tr></tr></tbody></table></center><input value="View Document" type="submit">"
>> document.forms[1].innerHTML
"<input name="caseid" value="605035" type="hidden"><input name="got_receipt" value="1" type="hidden">"

```

But I see the same with CMECF District now, which was not broken before, and maybe is with this code?:

```
>> document.forms.length
2
>> event.data.id
"form1516491978507"
>> document.forms
HTMLCollection [ <form>, <form#form1516491978507> ]
>> document.forms[0].outerHTML
"<form method="post" action="https://ecf.mad.uscourts.gov/doc1/09518488434" onsubmit="goDLS('/doc1/09518488434','192392','666','1','','1','','');return(false);"><hr><center><table width="400" border="1" bgcolor="white"><tbody><tr><th colspan="4"><font size="+1" color="DARKRED">Pacer Service Center </font></th></tr><tr><th colspan="4"><font color="DARKBLUE">Transaction Receipt </font></th></tr><tr></tr><tr></tr><tr><td colspan="4" align="CENTER"><font size="-1" color="DARKBLUE">Sat Jan 20 18:46:00 2018</font></td></tr><tr><th align="LEFT"><font size="-1" color="DARKBLUE"> Pacer Login: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> jhawkinson0:3355704:0 </font></td><th align="LEFT"><font size="-1" color="DARKBLUE"> Client Code: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE">  </font></td></tr><tr><th align="LEFT"><font size="-1" color="DARKBLUE"> Description: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> Image83-0 </font></td><th align="LEFT"><font size="-1" color="DARKBLUE"> Case Number: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> 1:17-cv-11842-PBS</font></td></tr><tr><th align="LEFT"><font size="-1" color="DARKBLUE"> Billable Pages: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> 2 </font></td><th align="LEFT"><font size="-1" color="DARKBLUE"> Cost: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> 0.20 </font></td></tr><tr></tr><tr></tr></tbody></table></center><input value="View Document" type="submit"></form>"
>> document.forms[1].outerHTML
"<form action="/doc1/09518488434" enctype="multipart/form-data" method="post" id="form1516491978507"><input name="caseid" value="192392" type="hidden"><input name="de_seq_num" value="666" type="hidden"><input name="got_receipt" value="1" type="hidden"><input name="pdf_toggle_possible" value="1" type="hidden"></form>"
```

Reply:
Whoops, correction, although it may not matter. The above BK example was from an image document in the docket report, but the proper test case would be from a claim document in the claims register. It seems the same with respect to which form is which, though:

```
>> document.forms
HTMLCollection [ <form>, <form#form1516523730586> ]
>> document.forms[0].outerHTML
"<form method="POST" action="https://ecf.caeb.uscourts.gov/doc1/032126913922" onsubmit="goDLS('/doc1/032126913922','605035','','1','2','','','43976411','70-1','');return(false)"><hr><center><table width="400" border="1" bgcolor="white"><tbody><tr><th colspan="4"><font size="+1" color="DARKRED">PACER Service Center </font></th></tr><tr><th colspan="4"><font color="DARKBLUE">Transaction Receipt </font></th></tr><tr></tr><tr></tr><tr><td colspan="4" align="CENTER"><font size="-1" color="DARKBLUE">Sun Jan 21 00:34:29 2018</font></td></tr><tr><th align="LEFT"><font size="-1" color="DARKBLUE"> Pacer Login: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> jhawkinson0:3355704:0 </font></td><th align="LEFT"><font size="-1" color="DARKBLUE"> Client Code: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE">  </font></td></tr><tr><th align="LEFT"><font size="-1" color="DARKBLUE"> Description: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> Claim 43976411-0 </font></td><th align="LEFT"><font size="-1" color="DARKBLUE"> Case Number: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> 17-13797                       </font></td></tr><tr><th align="LEFT"><font size="-1" color="DARKBLUE"> Billable Pages: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> 30 </font></td><th align="LEFT"><font size="-1" color="DARKBLUE"> Cost: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> 3.00 </font></td></tr><tr></tr><tr></tr></tbody></table></center><input value="View Document" type="submit"></form>"
>> document.forms[1].outerHTML
"<form action="/doc1/032126913922" method="post" id="form1516523730586"><input name="caseid" value="605035" type="hidden"><input name="got_receipt" value="1" type="hidden"><input name="pdf_header" value="2" type="hidden"><input name="claim_id" value="43976411" type="hidden"><input name="claim_num" value="70-1" type="hidden"></form>"
```

Maybe it's wise to make that more readable:
```html
<form method="POST" action="https://ecf.caeb.uscourts.gov/doc1/032126913922" onsubmit="goDLS('/doc1/032126913922','605035','','1','2','','','43976411','70-1','');return(false)">
<hr><center>
<table width="400" border="1" bgcolor="white"><tbody>
<tr><th colspan="4"><font size="+1" color="DARKRED">PACER Service Center </font></th></tr>
<tr><th colspan="4"><font color="DARKBLUE">Transaction Receipt </font></th></tr>
<tr></tr>
<tr></tr>
<tr><td colspan="4" align="CENTER"><font size="-1" color="DARKBLUE">Sun Jan 21 00:34:29 2018</font></td></tr>
<tr><th align="LEFT"><font size="-1" color="DARKBLUE"> Pacer Login: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> jhawkinson0:3355704:0 </font></td>
<th align="LEFT"><font size="-1" color="DARKBLUE"> Client Code: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE">  </font></td></tr>
<tr><th align="LEFT"><font size="-1" color="DARKBLUE"> Description: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> Claim 43976411-0 </font>
</td><th align="LEFT"><font size="-1" color="DARKBLUE"> Case Number: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> 17-13797                       </font></td></tr>
<tr><th align="LEFT"><font size="-1" color="DARKBLUE"> Billable Pages: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> 30 </font></td>
<th align="LEFT"><font size="-1" color="DARKBLUE"> Cost: </font></th><td align="LEFT"><font size="-1" color="DARKBLUE"> 3.00 </font></td></tr>
<tr></tr>
<tr></tr>
</tbody></table></center>
<input value="View Document" type="submit">
</form>
```
and
```html
<form action="/doc1/032126913922" method="post" id="form1516523730586">
<input name="caseid" value="605035" type="hidden">
<input name="got_receipt" value="1" type="hidden">
<input name="pdf_header" value="2" type="hidden">
<input name="claim_id" value="43976411" type="hidden">
<input name="claim_num" value="70-1" type="hidden">
</form>
```

> To feel confident that this is right, I'd want to know that document.forms[0] and document.getElementById(event.data.id) resolve to the same form. I don't have that confidence from inspecting the code. And when I look at PACER forms with the DOM inspector, etc., I don't see an id property. And when I set a breakpoint at the top of onDocumentViewSubmit(), I also don't get that.

So, knowing they resolve to the same form may be too high a standard. It should be sufficient if both forms' `onsubmit` actions get triggered in nearly the same browser page history context.  I'm not sure that's a true statement either, but it seems an easier reach?

Reply:
I tested this and it worked fine, but I reviewed it again today and realized there was indeed a change in functionality that this introduced. I backed out the changes. Even if it works, I don't mean to be tinkering here right now.